.claude/rules/architectural-rules.md

```
@@ -335,6 +335,92 @@ Located in `@green-goods/shared/hooks/utils/`:
+logger.error("Failed to submit work", { error });
+```
+
+**Exceptions:** `console.error` in indexer event handlers is acceptable (no logger available in Envio runtime).
+
+---
+
+## Rule 13: Provider Nesting Order
+
+**Pattern:** Incorrect provider hierarchy causing initialization failures
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)